vendor/bat-native-ledger/src/bat/ledger/internal/legacy/bat_helper.cc

```diff
@@ -12,83 +12,78 @@
 #include "bat/ledger/ledger.h"
-#include "rapidjson/document.h"
-#include "rapidjson/error/en.h"
-#include "rapidjson/stringbuffer.h"
-#include "rapidjson/writer.h"
 #include "third_party/re2/src/re2/re2.h"
 #include "tweetnacl.h"  // NOLINT
 #include "url/gurl.h"
 
 namespace braveledger_bat_helper {
 
-using JsonWriter = rapidjson::Writer<rapidjson::StringBuffer>;
+namespace {
+constexpr char kFieldEvent[] = "event";
+constexpr char kFieldProperties[] = "properties";
+constexpr char kFieldChannel[] = "channel";
+constexpr char kFieldVod[] = "vod";
+constexpr char kFieldTime[] = "time";
+}  // namespace
 
 /////////////////////////////////////////////////////////////////////////////
 
-bool getJSONValue(const std::string& fieldName,
+bool getJSONValue(const std::string& field_name,
                   const std::string& json,
                   std::string* value) {
-  rapidjson::Document d;
-  d.Parse(json.c_str());
-
-  // has parser errors or wrong types
-  bool error = d.HasParseError() || (false == d.HasMember(fieldName.c_str()));
-  if (!error) {
-    *value = d[fieldName.c_str()].GetString();
+  absl::optional<base::Value> result =
+      base::JSONReader::Read(json, base::JSON_PARSE_CHROMIUM_EXTENSIONS |
+                                       base::JSONParserOptions::JSON_PARSE_RFC);
+  if (!result || !result->is_dict())
+    return false;
+
+  if (auto* field = result->GetDict().FindString(field_name)) {
+    *value = *field;
+    return true;
   }
-  return !error;
+
+  return false;
 }
 
 bool getJSONTwitchProperties(
     const std::string& json,
     std::vector<base::flat_map<std::string, std::string>>* parts) {
-  rapidjson::Document d;
-  d.Parse(json.c_str());
-
-  // has parser errors or wrong types
-  bool error = d.HasParseError();
-  if (!error) {
-    for (auto & i : d.GetArray()) {
-      const char * event_field = "event";
-      base::flat_map<std::string, std::string> eventmap;
-
-      auto obj = i.GetObject();
-      if (obj.HasMember(event_field)) {
-        eventmap[event_field] = obj[event_field].GetString();
-      }
-
-      const char * props_field = "properties";
-      if (obj.HasMember(props_field)) {
-        eventmap[props_field] = "";
-
-        const char * channel_field = "channel";
-        if (obj[props_field].HasMember(channel_field) &&
-          obj[props_field][channel_field].IsString()) {
-          eventmap[channel_field] = obj[props_field][channel_field].GetString();
-        }
-
-        const char * vod_field = "vod";
-        if (obj[props_field].HasMember(vod_field)) {
-          eventmap[vod_field] = obj[props_field][vod_field].GetString();
-        }
-
-        const char * time_field = "time";
-        if (obj[props_field].HasMember(time_field)) {
-          double d = obj[props_field][time_field].GetDouble();
-          eventmap[time_field] = std::to_string(d);
-        }
-      }
-      parts->push_back(eventmap);
+  absl::optional<base::Value> result =
```

Comment:
Same here.

Reply:
Fixed in e2c74682957f896d5481ee3e371a29b17b7b7929.